src/components/execution/AddExecutionForm.js

```diff
@@ -0,0 +1,288 @@
+  static propTypes = {
+    results: PropTypes.instanceOf(List),
+    datasets: PropTypes.instanceOf(List),
+    algorithms: PropTypes.instanceOf(List),
+    t: PropTypes.func.isRequired,
+    dispatchGetDatasets: PropTypes.func.isRequired,
+    dispatchGetAlgorithms: PropTypes.func.isRequired,
+    dispatchExecuteAlgorithm: PropTypes.func.isRequired,
+    dispatchGetResults: PropTypes.func.isRequired,
+    classes: PropTypes.shape({
+      formControl: PropTypes.string.isRequired,
+      infoAlert: PropTypes.string.isRequired,
+      buttonContainer: PropTypes.string.isRequired,
+      buttonWrapper: PropTypes.string.isRequired,
+      menuItem: PropTypes.string.isRequired,
+    }).isRequired,
+    isLoading: PropTypes.bool.isRequired,
+    pythonVersion: PropTypes.shape({
+      valid: PropTypes.bool,
+      version: PropTypes.string,
+    }).isRequired,
+  };
+
+  static defaultProps = {
+    datasets: null,
+    algorithms: null,
+    results: null,
+  };
+
+  state = {
+    sourceId: '',
+    algorithmId: '',
+    filename: '',
+  };
+
+  componentDidMount() {
+    const {
+      dispatchGetDatasets,
+      dispatchGetAlgorithms,
+      dispatchGetResults,
+    } = this.props;
+    dispatchGetDatasets();
+    dispatchGetAlgorithms();
+    dispatchGetResults();
+  }
+
+  handleDatasetSelectOnChange = (e) => {
+    this.setState({ sourceId: e.target.value });
+  };
+
+  handleAlgorithmSelectOnChange = (e) => {
+    this.setState({ algorithmId: e.target.value });
+  };
+
+  handleNameOnChange = (e) => {
+    this.setState({ filename: e.target.value });
+  };
+
+  executeAlgorithm = () => {
+    const { dispatchExecuteAlgorithm, algorithms } = this.props;
+    const { sourceId, algorithmId, filename } = this.state;
+    const { language } = algorithms.find(({ id }) => id === algorithmId);
+    if (language) {
+      dispatchExecuteAlgorithm({ sourceId, algorithmId, language, filename });
+    }
+  };
+
+  renderDatasetsAndResultsSelect = () => {
+    const { sourceId } = this.state;
+    const { classes, t, datasets, results } = this.props;
+
+    if (datasets?.isEmpty() || results?.isEmpty()) {
+      return <Loader />;
+    }
+
+    const datasetMenuItems = datasets
+      .sortBy(({ name }) => name)
+      .map(({ id, name }) => (
+        <MenuItem value={id} key={id} className={classes.menuItem}>
+          {name}
+        </MenuItem>
+      ));
+
+    const resultMenuItems = results
+      .sortBy(({ name }) => name)
+      .map(({ id, name }) => (
+        <MenuItem value={id} key={id} className={classes.menuItem}>
+          {name}
+        </MenuItem>
+      ));
+
+    return (
+      <FormControl variant="outlined" className={classes.formControl}>
+        <InputLabel id="dataset-select">
+          {`${t('Dataset')} ${t('(Required)')}`}
+        </InputLabel>
+        <Select
+          labelId="dataset-select"
+          value={sourceId}
+          onChange={this.handleDatasetSelectOnChange}
+          label={t('Dataset')}
```

Comment:
The issue @hagoptaminian mentioned where `(required)` is crossed out is caused by this `label` prop value not corresponding to the one in the `InputLabel`. Replace `t('Dataset')` by `` `${t('Dataset')} ${t('(Required)')}` ``.